depclean-gradle-plugin/src/main/java/se/kth/depclean/analysis/DefaultGradleProjectDependencyAnalyzer.java

```diff
@@ -0,0 +1,345 @@
+      Set<ResolvedArtifact> usedDeclaredArtifacts = new LinkedHashSet<>(declaredArtifacts);
+      usedDeclaredArtifacts.retainAll(usedArtifacts);
+
+      // for the used dependencies, remove the ones that are declared
+      Set<ResolvedArtifact> usedUndeclaredArtifacts = new LinkedHashSet<>(usedArtifacts);
+      usedUndeclaredArtifacts = removeAll(usedUndeclaredArtifacts, declaredArtifacts);
+
+      // for the declared dependencies, get the ones that are not used
+      Set<ResolvedArtifact> unusedDeclaredArtifacts = new LinkedHashSet<>(declaredArtifacts);
+      unusedDeclaredArtifacts = removeAll(unusedDeclaredArtifacts, usedArtifacts);
+
+      return new GradleProjectDependencyAnalysis(usedDeclaredArtifacts,
+              usedUndeclaredArtifacts, unusedDeclaredArtifacts);
+
+    } catch (IOException e) {
+      throw new ProjectDependencyAnalyzerException("Cannot analyze dependencies", e);
+    }
+  }
+
+  /**
+   * Returns a map with the artifacts (dependencies) in a Gradle project and
+   * their corresponding classes.
+   *
+   * @param allArtifacts File of each artifact.
+   * @return A map of artifact -> classes.
+   * @throws IOException If the class cannot be analyzed.
+   */
+  public Map<ResolvedArtifact, Set<String>> buildArtifactClassMap(
+          final Set<ResolvedArtifact> allArtifacts) throws IOException {
+    Map<ResolvedArtifact, Set<String>> artifactClassMap = new LinkedHashMap<>();
+    for (ResolvedArtifact artifact : allArtifacts) {
+      File file = artifact.getFile();
+      if (file.getName().endsWith(".jar")) {
+        // optimized solution for the jar case
+        try (JarFile jarFile = new JarFile(file)) {
+          Enumeration<JarEntry> jarEntries = jarFile.entries();
+          Set<String> classes = new HashSet<>();
+          while (jarEntries.hasMoreElements()) {
+            String entry = jarEntries.nextElement().getName();
+            if (entry.endsWith(".class")) {
+              String className = entry.replace('/', '.');
+              className = className.substring(0, className.length() - ".class".length());
+              classes.add(className);
+            }
+          }
+          artifactClassMap.put(artifact, classes);
+        }
+      } else if (file.isDirectory()) {
+        URL url = file.toURI().toURL();
+        Set<String> classes = classAnalyzer.analyze(url);
+        artifactClassMap.put(artifact, classes);
+      }
+    }
+    return artifactClassMap;
+  }
+
+  /**
+   * Get the project's build classes.
+   *
+   * @param project The gradle project.
+   * @throws IOException In case of IO issues.
+   */
+  private void  buildProjectDependencyClasses(final Project project) throws IOException {
+    String sep = File.separator;
+    String classesDir = project.getProjectDir().getAbsolutePath()
+            + sep + "build"
+            + sep + "classes";
+
+    // Analyze src classes in the project
+    File outputJavaDir = new File(classesDir + sep + "java" + sep + "main");
```

Comment:
You could use Paths.get: https://docs.oracle.com/javase/7/docs/api/java/nio/file/Paths.html#get(java.lang.String,%20java.lang.String...)